include/sddf/serial/queue.h

```diff
@@ -167,27 +167,20 @@ static inline void serial_update_visible_head(serial_queue_handle_t *queue_handl
     queue_handle->queue->head = local_head;
 }
 
-static inline uint32_t serial_queue_contiguous_length(serial_queue_handle_t *queue_handle)
+static inline uint32_t serial_queue_length(serial_queue_handle_t *queue_handle)
 {
-    uint32_t head = queue_handle->queue->head;
-    uint32_t tail = queue_handle->queue->tail;
-
-    if (head <= tail) {
-        return tail - head;
-    } else {
-        return (UINT32_MAX - head) + 1;
-    }
+    return queue_handle->queue->tail - queue_handle->queue->head;
 }
 
-static inline uint32_t serial_queue_length(serial_queue_handle_t *queue_handle)
+static inline uint32_t serial_queue_contiguous_length(serial_queue_handle_t *queue_handle)
 {
-    uint32_t head = queue_handle->queue->head;
-    uint32_t tail = queue_handle->queue->tail;
+    uint32_t head = queue_handle->queue->head % queue_handle->size;
+    uint32_t tail = queue_handle->queue->tail % queue_handle->size;
 
     if (head <= tail) {
-        return serial_queue_contiguous_length(queue_handle);
+        return tail - head;
     } else {
-        return serial_queue_contiguous_length(queue_handle) + tail;
+        return queue_handle->size - head;
```

Comment:
Would `(tail - head) % size` be simpler? That's what the other headers do.

Reply:
Nope, because that's not contiguous space.  This function needs a comment as to what it's returning.  Is it the amount of space that can be filled with a memcpy? (head - tail or size-tail) OR the amount of contiguous filled space (size - (head %size)) or `(tail-head)`

Either way there are extra modulo operations in here that aren't necessary.

Reply:
I have improved the implementations of all the contiguous free/length functions, and added descriptions to every functions. I've also improved some inconsistent descriptions as well.